### ISSUE TITLE
R CMD check fails on baselineWavelet.Rd

### ISSUE THREAD

**Issue by FMKerckhof** [zmzhang/baselineWavelet#1]
Dear,

I am trying to install the baselineWavelet in R 3.6.0 running on ubuntu linux 18.04. 
While trying to build the R package from github, R CMD check fails with the following significant warning:

`<...>/baselineWavelet.Rcheck/00_pkg_src/baselineWavelet/man/baselineWavelet.Rd: docType ‘baselineWavelet’ is unrecognized`

I don't really know how to bypass this, I always document with roxygen. Any suggestions?

My full R CMD check log:

```
* using R version 3.6.0 (2019-04-26)
* using platform: x86_64-pc-linux-gnu (64-bit)
* using session charset: UTF-8
* checking for file ‘baselineWavelet/DESCRIPTION’ ... OK
* checking extension type ... Package
* this is package ‘baselineWavelet’ version ‘4.0.2’
* checking package namespace information ... OK
* checking package dependencies ... OK
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... OK
* checking for portable file names ... OK
* checking for sufficient/correct file permissions ... OK
* checking whether package ‘baselineWavelet’ can be installed ... WARNING
Found the following significant warnings:
  Warning: <..>/baselineWavelet.Rcheck/00_pkg_src/baselineWavelet/man/baselineWavelet.Rd: docType ‘baselineWavelet’ is unrecognized
See ‘<...>/baselineWavelet.Rcheck/00install.out’ for details.
* checking installed package size ... OK
* checking package directory ... OK
* checking DESCRIPTION meta-information ... NOTE
Malformed Description field: should contain one or more complete sentences.
Non-standard license specification:
  LGPL version 2 or newer
Standardizable: TRUE
Standardized license specification:
  LGPL (>= 2.0)
* checking top-level files ... OK
* checking for left-over files ... OK
* checking index information ... OK
* checking package subdirectories ... OK
* checking R files for non-ASCII characters ... OK
* checking R files for syntax errors ... OK
* checking whether the package can be loaded ... OK
* checking whether the package can be loaded with stated dependencies ... OK
* checking whether the package can be unloaded cleanly ... OK
* checking whether the namespace can be loaded with stated dependencies ... OK
* checking whether the namespace can be unloaded cleanly ... OK
* checking loading without being on the library search path ... OK
* checking dependencies in R code ... OK
* checking S3 generic/method consistency ... OK
* checking replacement functions ... OK
* checking foreign function calls ... OK
* checking R code for possible problems ... NOTE
WhittakerSmooth: no visible global function definition for ‘as’
cwt: no visible global function definition for ‘convolve’
extendNBase: no visible global function definition for ‘nextn’
identifyMajorPeaks: no visible global function definition for
  ‘quantile’
identifyMajorPeaks: no visible global function definition for ‘sd’
identifyMajorPeaks: no visible global function definition for ‘mad’
peakDetectionCWT: no visible global function definition for ‘sav.gol’
peakDetectionCWT: no visible global function definition for
  ‘tuneInPeakInfo’
plotRidgeList: no visible global function definition for
  ‘colorRampPalette’
plotRidgeList: no visible global function definition for ‘plot’
plotRidgeList: no visible global function definition for ‘points’
plotwidthEstimation: no visible global function definition for ‘points’
plotwidthEstimation: no visible global function definition for ‘text’
Undefined global functions or variables:
  as colorRampPalette convolve mad nextn plot points quantile sav.gol
  sd text tuneInPeakInfo
Consider adding
  importFrom("grDevices", "colorRampPalette")
  importFrom("graphics", "plot", "points", "text")
  importFrom("methods", "as")
  importFrom("stats", "convolve", "mad", "nextn", "quantile", "sd")
to your NAMESPACE file (and ensure that your DESCRIPTION Imports field
contains 'methods').
* checking Rd files ... WARNING
prepare_Rd: baselineWavelet.Rd: docType ‘baselineWavelet’ is unrecognized
* checking Rd metadata ... OK
* checking Rd cross-references ... OK
* checking for missing documentation entries ... WARNING
Undocumented data sets:
  ‘m’ ‘xa’
All user-level objects in a package should have documentation entries.
See chapter ‘Writing R documentation files’ in the ‘Writing R
Extensions’ manual.
* checking for code/documentation mismatches ... WARNING
Functions or methods with usage in documentation object 'getRidgeLength' but not in code:
  ‘getRidgeLength’

Functions or methods with usage in documentation object 'getRidgeValue' but not in code:
  ‘getRidgeValue’

Codoc mismatches from documentation object 'WhittakerSmooth':
WhittakerSmooth
  Code: function(x, w, lambda, differences = 1)
  Docs: function(x, w, lambda)
  Argument names in code not in docs:
    differences

Codoc mismatches from documentation object 'baselineCorrectionCWT':
baselineCorrectionCWT
  Code: function(x, peakWidth, threshold = 0.5, lambda = 100,
                 differences = 1)
  Docs: function(x, peakWidth, threshold = 0.5, lambda = 100)
  Argument names in code not in docs:
    differences

Codoc mismatches from documentation object 'identifyMajorPeaks':
identifyMajorPeaks
  Code: function(ms, ridgeList, wCoefs, scales =
                 as.numeric(colnames(wCoefs)), SNR.Th = 1,
                 peakScaleRange = 5, ridgeLength = 10, nearbyPeak =
                 FALSE, nearbyWinSize = 100, winSize.noise = 500,
                 SNR.method = "quantile", minNoiseLevel = 0.001)
  Docs: function(ms, ridgeList, wCoefs, scales =
                 as.numeric(colnames(wCoefs)), SNR.Th = 3,
                 peakScaleRange = 5, ridgeLength = 32, nearbyPeak =
                 FALSE, nearbyWinSize = 100, winSize.noise = 500,
                 SNR.method = "quantile", minNoiseLevel = 0.001)
  Mismatches in argument default values:
    Name: 'SNR.Th' Code: 1 Docs: 3
    Name: 'ridgeLength' Code: 10 Docs: 32

* checking Rd \usage sections ... WARNING
Documented arguments not in \usage in documentation object 'WhittakerSmooth':
  ‘differences’

Documented arguments not in \usage in documentation object 'baselineCorrectionCWT':
  ‘differences’

Functions with \usage entries need to have the appropriate \alias
entries, and all their arguments documented.
The \usage entries must correspond to syntactically valid R code.
See chapter ‘Writing R documentation files’ in the ‘Writing R
Extensions’ manual.
* checking Rd contents ... OK
* checking for unstated dependencies in examples ... OK
* checking contents of ‘data’ directory ... OK
* checking data for non-ASCII characters ... OK
* checking data for ASCII and uncompressed saves ... OK
* checking examples ... ERROR
Running examples in ‘baselineWavelet-Ex.R’ failed
The error most likely occurred in:

> ### Name: getLocalMaximumCWT
> ### Title: Identify the local maximum of each column in 2-D CWT
> ###   coefficients matrix
> ### Aliases: getLocalMaximumCWT
> ### Keywords: methods
> 
> ### ** Examples
> 
> 	data(exampleMS)
> 	scales <- seq(1, 64, 3)
> 	wCoefs <- cwt(exampleMS[5000:11000], scales=scales, wavelet='mexh')
> 	
> 	localMax <- getLocalMaximumCWT(wCoefs)
> 	plotLocalMax(localMax)
Error in plotLocalMax(localMax) : could not find function "plotLocalMax"
Execution halted
* checking PDF version of manual ... OK
* DONE
Status: 1 ERROR, 5 WARNINGs, 2 NOTEs
```

**Comment by zmzhang** [comment]
@FMKerckhof 

R CMD check fails because of following issues:
1. we used some inexistent functions in the examples.
2. Some necessary functions were not exported too.

In commit [a5cad6b](https://github.com/zmzhang/baselineWavelet/commit/a5cad6b77a97e3e43b300e26351aa172d5d3e6b9), we have fixed these issues. All the checks can be passed smoothly.

![pass checks](https://user-images.githubusercontent.com/6937141/57864286-3389ac00-782e-11e9-8796-3baa60b147ee.png)
